commit 4041daafc89c24c065ac48ee00d01a3d8d0f09b0 (refs/changes/94/5494/2)
Author: Robert Mustacchi <rm@joyent.com>
Date:   2019-02-08T16:25:05+00:00 (8 months ago)
    
    OS-7565 drop support for old CTF tools from illumos-extra

diff --git a/Makefile b/Makefile
index a9e261c..ab41c95 100644
--- a/Makefile
+++ b/Makefile
@@ -172,7 +172,6 @@ $(SUBDIRS): $(DESTDIR)/usr/gcc/$(PRIMARY_COMPILER_VER)/bin/gcc
 	    STRAP=$(STRAP) \
 	    CTFMERGE=$(CTFMERGE) \
 	    CTFCONVERT=$(CTFCONVERT) \
-	    ALTCTFCONVERT=$(ALTCTFCONVERT) \
 	    $(MAKE) DESTDIR=$(DESTDIR) install)
 
 $(STRAPFIX_SUBDIRS): $(SUBDIRS)
@@ -207,7 +206,6 @@ $(SUBDIRS): $(PRIMARY_COMPILER)
 	    STRAP=$(STRAP) \
 	    CTFMERGE=$(CTFMERGE) \
 	    CTFCONVERT=$(CTFCONVERT) \
-	    ALTCTFCONVERT=$(ALTCTFCONVERT) \
 	    $(MAKE) DESTDIR=$(DESTDIR) install)
 
 install: $(PRIMARY_COMPILER) $(SUBDIRS)
diff --git a/mdb_v8/Makefile b/mdb_v8/Makefile
index 94ff7fe..5c60cac 100644
--- a/mdb_v8/Makefile
+++ b/mdb_v8/Makefile
@@ -10,7 +10,7 @@
 #
 
 #
-# Copyright (c) 2018, Joyent, Inc.
+# Copyright (c) 2019, Joyent, Inc.
 #
 
 #
@@ -55,10 +55,10 @@ all: $(VER)/GNUMakefile
 
 install: all
 	rm -f $(DESTDIR)/usr/lib/mdb/proc/v8.so
-	$(ALTCTFCONVERT) -l $(VER) -o $(DESTDIR)/usr/lib/mdb/proc/v8.so \
+	$(CTFCONVERT) -l $(VER) -o $(DESTDIR)/usr/lib/mdb/proc/v8.so \
 	    $(VER)/build/ia32/mdb_v8.so
 	rm -f $(DESTDIR)/usr/lib/mdb/proc/amd64/v8.so
-	$(ALTCTFCONVERT) -l $(VER) -o $(DESTDIR)/usr/lib/mdb/proc/amd64/v8.so \
+	$(CTFCONVERT) -l $(VER) -o $(DESTDIR)/usr/lib/mdb/proc/amd64/v8.so \
 	    $(VER)/build/amd64/mdb_v8.so
 
 $(VER)/GNUMakefile: $(VER).tar.gz
diff --git a/ntp/Makefile b/ntp/Makefile
index 4742244..0ac92cb 100644
--- a/ntp/Makefile
+++ b/ntp/Makefile
@@ -19,7 +19,7 @@
 # CDDL HEADER END
 #
 # Copyright (c) 2009, 2010, Oracle and/or its affiliates. All rights reserved.
-# Copyright 2016 Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 VER =	ntp-4.2.8p8
@@ -59,9 +59,7 @@ CFLAGS += \
 AUTOCONF_ENV += \
 	LD_OPTIONS="$(LDFLAGS)" \
 	DESTDIR="$(DESTDIR)" \
-	CTFMERGE="$(CTFMERGE)" \
-	CTFCONVERT="$(CTFCONVERT)" \
-	ALTCTFCONVERT="$(ALTCTFCONVERT)"
+	CTFCONVERT="$(CTFCONVERT)"
 
 PATCHSTRIP =	0
 PATCHES =	Patches/*
diff --git a/openssh/Makefile b/openssh/Makefile
index 0f8afc6..8f7939e 100644
--- a/openssh/Makefile
+++ b/openssh/Makefile
@@ -19,7 +19,7 @@
 # CDDL HEADER END
 #
 # Copyright (c) 2009, 2010, Oracle and/or its affiliates. All rights reserved.
-# Copyright 2015 Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 VER =	openssh-portable-7.4p1
@@ -53,9 +53,7 @@ AUTOCONF_OPTS += \
 
 AUTOCONF_ENV += \
 	DESTDIR="$(DESTDIR)" \
-	CTFMERGE="$(CTFMERGE)" \
-	CTFCONVERT="$(CTFCONVERT)" \
-	ALTCTFCONVERT="$(ALTCTFCONVERT)"
+	CTFCONVERT="$(CTFCONVERT)"
 
 CFLAGS += \
 	-O2 -gdwarf-2 \
diff --git a/tools/make-ctf b/tools/make-ctf
index 845355b..918fea6 100755
--- a/tools/make-ctf
+++ b/tools/make-ctf
@@ -1,5 +1,18 @@
 #!/bin/ksh
-# vim: set ts=8 sts=8 sw=8 noet:
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2019, Joyent, Inc.
+#
 
 set -o errexit
 set -o pipefail
@@ -34,61 +47,24 @@ if [[ -z "${DESTDIR}" ]]; then
 fi
 
 # These may contain wildcards, so expand:
-CTFMERGE="$(echo $CTFMERGE)"
 CTFCONVERT="$(echo $CTFCONVERT)"
-ALTCTFCONVERT="$(echo $ALTCTFCONVERT)"
-if [[ ! -n "${CTFMERGE}" || ! -n "${CTFCONVERT}" ]]; then
-	printf "You must set CTFMERGE and CTFCONVERT in the environment.\n" >&2
-	exit 1
-fi
-if [[ ! -x "${CTFMERGE}" ]]; then
-	printf "Could not find CTFMERGE at '%s'\n" "${CTFMERGE}" >&2
+if [[ ! -n "${CTFCONVERT}" ]]; then
+	printf "You must set CTFCONVERT in the environment.\n" >&2
 	exit 1
 fi
+
 if [[ ! -x "${CTFCONVERT}" ]]; then
 	printf "Could not find CTFCONVERT at '%s'\n" "${CTFCONVERT}" >&2
 	exit 1
 fi
-if [[ -n "${ALTCTFCONVERT}" ]]; then
-	if [[ ! -x "${ALTCTFCONVERT}" ]]; then
-		printf "Could not find ALTCTFCONVERT at '%s'\n" \
-		    "${ALTCTFCONVERT}" >&2
-		exit 1
-	fi
-	NEW_CTF_TOOLS=yes
-else
-	NEW_CTF_TOOLS=no
-fi
-
-if [[ $NEW_CTF_TOOLS != 'yes' ]]; then
-	# Convert DWARF to CTF in every object file we can find:
-	rm -rf "${CTF_OBJECT_DIR}"
-	mkdir -p "${CTF_OBJECT_DIR}"
-	(cd "${BUILD_DIR}" && find * -name '*.o' -type f) | \
-	    sort | while read objf; do
-		printf "  CTFCONVERT %s\n" "${objf}" >&2
-		ctff="$(echo "$objf" | sed -e 's,/,%,g' -e 's/\.o$/\.ctf/')"
-		${CTFCONVERT} -l "${CONTAINER_VERSION}" \
-		    -o "${CTF_OBJECT_DIR}/${ctff}" \
-		    "${BUILD_DIR}/${objf}"
-	done
-fi
 
 # Merge the CTF into the set of programs we ship:
 for program in ${PROGRAMS}; do
 	program_path="${DESTDIR}${program}"
-	if [[ $NEW_CTF_TOOLS == 'yes' ]]; then
-		printf "  ALTCTFCONV %s\n" "${program_path}" >&2
-		chmod u+w "${program_path}"
-		${ALTCTFCONVERT} -l "${CONTAINER_VERSION}" \
-		    -o "${program_path}" "${program_path}"
-	else
-		printf "  CTFMERGE   %s\n" "${program_path}" >&2
-		chmod u+w "${program_path}"
-		${CTFMERGE} -l "${CONTAINER_VERSION}" \
-		    -o "${program_path}" \
-		    ${CTF_OBJECT_DIR}/*.ctf
-	fi
+	printf "  CTFCONV %s\n" "${program_path}" >&2
+	chmod u+w "${program_path}"
+	${CTFCONVERT} -l "${CONTAINER_VERSION}" \
+	    -o "${program_path}" "${program_path}"
 	if [[ -z $NO_STRIP_SOURCE_DEBUG ]]; then
 		printf "  STRIP      %s\n" "${program_path}" >&2
 		/usr/bin/strip -x "${program_path}"
diff --git a/wget/Makefile b/wget/Makefile
index 450dcef..277e31b 100644
--- a/wget/Makefile
+++ b/wget/Makefile
@@ -10,7 +10,7 @@
 #
 
 #
-# Copyright (c) 2018, Joyent, Inc.
+# Copyright (c) 2019, Joyent, Inc.
 #
 
 VER =			wget-1.18
@@ -24,9 +24,7 @@ AUTOCONF_ENV += \
 			PERL=/opt/local/bin/perl \
 			POD2MAN=/opt/local/bin/pod2man \
 			DESTDIR="$(DESTDIR)" \
-			CTFMERGE="$(CTFMERGE)" \
-			CTFCONVERT="$(CTFCONVERT)" \
-			ALTCTFCONVERT="$(ALTCTFCONVERT)"
+			CTFCONVERT="$(CTFCONVERT)"
 
 AUTOCONF_OPTS += \
 			--sysconfdir=/etc \
